sensor/common/clusterentities/store_ips.go

```diff
@@ -0,0 +1,217 @@
+	e.mutex.Lock()
+	defer e.mutex.Unlock()
+	// Maps holding historical data must not be wiped on reset! Instead, all entities must be marked as historical.
+	// Must be called before the respective source maps are wiped!
+	// Performance optimization: no need to handle history if history is disabled
+	if e.entitiesMemorySize == 0 {
+		e.initMapsNoLock()
+		return
+	}
+	e.markAllIPsHistorical()
+
+	e.ipMap = make(map[net.IPAddress]map[string]struct{})
+	e.reverseIPMap = make(map[string]map[net.IPAddress]struct{})
+	e.updateMetricsNoLock()
+}
+
+func (e *ipsStore) updateMetricsNoLock() {
+}
+
+func (e *ipsStore) RecordTick() {
+	for ip, m := range e.historicalIPs {
+		for deploymentID, status := range m {
+			status.recordTick()
+			// Remove all historical entries that expired in this tick.
+			e.removeHistoricalExpiredIPs(deploymentID, ip)
+		}
+	}
+}
+
+func (e *ipsStore) Apply(
+	updates map[string]*EntityData,
+	incremental bool,
+	incPublicIPRefNoLock func(addr net.IPAddress),
+	decPublicIPRefNoLock func(addr net.IPAddress),
+) {
+	e.mutex.Lock()
+	defer e.mutex.Unlock()
+	e.applyNoLock(updates, incremental, incPublicIPRefNoLock, decPublicIPRefNoLock)
+}
+
+func (e *ipsStore) applyNoLock(updates map[string]*EntityData, incremental bool,
+	incPublicIPRefNoLock func(addr net.IPAddress), decPublicIPRefNoLock func(addr net.IPAddress),
+) {
+	defer e.updateMetricsNoLock()
+	if !incremental {
+		for deploymentID := range updates {
+			e.purgeNoLock(deploymentID, decPublicIPRefNoLock)
+		}
+	}
+	for deploymentID, data := range updates {
+		if data == nil {
+			continue
+		}
+		e.applySingleNoLock(deploymentID, *data, incPublicIPRefNoLock)
+	}
+}
+
+func (e *ipsStore) purgeNoLock(deploymentID string, decPublicIPRefNoLock func(addr net.IPAddress)) {
+	for ip := range e.reverseIPMap[deploymentID] {
+		e.markHistoricalIP(deploymentID, ip)
+		// For entitiesMemorySize > 0, the deletion of historical expired entries happens after a tick.
+		// If memory is disabled, we should not wait for a tick and delete them immediately.
+		e.removeHistoricalExpiredIPs(deploymentID, ip)
+
+		delete(e.ipMap, ip)
+		if ip.IsPublic() {
+			decPublicIPRefNoLock(ip)
+		}
+	}
+	delete(e.reverseIPMap, deploymentID)
+}
+
+func (e *ipsStore) LookupByNetAddr(ip net.IPAddress, port uint16) (results []LookupResult) {
+	for deploymentID := range e.ipMap[ip] {
+		result := LookupResult{
+			Entity:         networkgraph.EntityForDeployment(deploymentID),
+			ContainerPorts: []uint16{port},
+		}
+		results = append(results, result)
+	}
+	// if there is a match in the map, then there is no need to search in history,
+	// as it may contain data about different past deployment using this address
+	if len(results) > 0 {
+		return results
+	}
+	for histDeploymentID := range e.historicalIPs[ip] {
+		result := LookupResult{
+			Entity:         networkgraph.EntityForDeployment(histDeploymentID),
+			ContainerPorts: []uint16{port},
+		}
+		results = append(results, result)
+	}
```

Comment:
if I understand this correctly, this is necessary because we unconditionally delete from the `ipMap` in `purgeNoLock` but a similar code it's not needed in the endpoint store because we do not delete unconditionally from the `endpointMap` and the `endpointMap` will store the historical data until expires, right?